src/qutip_qip/circuit.py

```diff
@@ -1262,240 +1262,64 @@ def propagators(self, expand=True):
-            elif gate.name == "Y":
-                U_list.append(y_gate())
-            elif gate.name == "CY":
-                U_list.append(cy_gate())
-            elif gate.name == "Z":
-                U_list.append(z_gate())
-            elif gate.name == "CZ":
-                U_list.append(cz_gate())
-            elif gate.name == "T":
-                U_list.append(t_gate())
-            elif gate.name == "CT":
-                U_list.append(ct_gate())
-            elif gate.name == "S":
-                U_list.append(s_gate())
-            elif gate.name == "CS":
-                U_list.append(cs_gate())
-            elif gate.name == "SQRTNOT":
-                U_list.append(sqrtnot())
-            elif gate.name == "SNOT":
-                U_list.append(snot())
-            elif gate.name == "PHASEGATE":
-                U_list.append(phasegate(gate.arg_value))
-            elif gate.name == "QASMU":
-                U_list.append(qasmu_gate(gate.arg_value))
-            elif gate.name == "CRX":
-                U_list.append(controlled_gate(rx(gate.arg_value)))
-            elif gate.name == "CRY":
-                U_list.append(controlled_gate(ry(gate.arg_value)))
-            elif gate.name == "CRZ":
-                U_list.append(controlled_gate(rz(gate.arg_value)))
-            elif gate.name == "CPHASE":
-                U_list.append(cphase(gate.arg_value))
-            elif gate.name == "CNOT":
-                U_list.append(cnot())
-            elif gate.name == "CSIGN":
-                U_list.append(csign())
-            elif gate.name == "BERKELEY":
-                U_list.append(berkeley())
-            elif gate.name == "SWAPalpha":
-                U_list.append(swapalpha(gate.arg_value))
-            elif gate.name == "SWAP":
-                U_list.append(swap())
-            elif gate.name == "ISWAP":
-                U_list.append(iswap())
-            elif gate.name == "SQRTSWAP":
-                U_list.append(sqrtswap())
-            elif gate.name == "SQRTISWAP":
-                U_list.append(sqrtiswap())
-            elif gate.name == "FREDKIN":
-                U_list.append(fredkin())
-            elif gate.name == "TOFFOLI":
-                U_list.append(toffoli())
-            elif gate.name == "GLOBALPHASE":
-                U_list.append(globalphase(gate.arg_value, self.N))
-            elif gate.name == "IDLE":
-                U_list.append(qeye(2))
-            elif gate.name in self.user_gates:
+            if gate.name == "GLOBALPHASE":
+                qobj = gate.get_qobj(self.N)
+                U_list.append(qobj)
+                continue
+
+            if gate.name in self.user_gates:
                 if gate.controls is not None:
                     raise ValueError("A user defined gate {} takes only  "
-                                     "`targets` variable.".format(gate.name))
+                                    "`targets` variable.".format(gate.name))
```

Comment:
Not sure why the indentation was changed here.

Reply:
Hmm, don't know. Removed.